### PR TITLE
fix ucs catalog test and enable for smoke test

### PR DESCRIPTION
**Pull request by geoff-reid**
Fix for https://rackhd.atlassian.net/browse/RAC-4559

The fit_comon.rackhdapi() routine uses an empty array as the default payload.  The 2.0/nodes/id/workflows API requires an object as a payload instead of an array.  The fix is to pass an empty object to the routine.  

@keedya @uppalk1 @tannoa2 @RackHD/corecommitters 

removing "depends on" since the fix has been merged into on-tasks

### CODE REVIEW

**Comment by panpan0000** [comment]
test this please

**Comment by hohene** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by hohene** [comment]
@geoff-reid seems all your submodules are listed in this PR now?  Not sure what is getting pulled in for the test build.

**Comment by geoff-reid** [comment]
Test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by keedya** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by geoff-reid** [comment]
@hohene, I finally got the UCS tests to pass, can you take a look at this PR?  Thanks

File: test/tests/ucs/test_rackhd_api_ucs.py [hunk 1]
```diff
@@ -194,28 +228,24 @@ def test_api_20_ucs_discovery(self):
...
         Tests the UCS Catalog workflow in rackHD
         :return:
         """
-        catalog_workflows = []
         for x in range(len(self.UCS_NODES)):
             postUrl = '/api/2.0/nodes/' + str(self.UCS_NODES[x]["id"]) + "/workflows?name=Graph.Ucs.Catalog"
             header = {"Content-Type": "application/json"}
-            api_data = fit_common.rackhdapi(postUrl, headers=header, action="post")
+            api_data = fit_common.rackhdapi(postUrl, headers=header, action="post", payload={})
             self.assertEqual(api_data['status'], 201,
```
Comment (hohene) [inline test/tests/ucs/test_rackhd_api_ucs.py]: Suggestion for later.  This test will assert on the first error found.  For testing, may want to build a list of errors and walk all the USCnodes to see if any worked for status/cataloging.  Similar to what Larry did in this test script -  test/tests/ha/rackdh_ha_resource_test.py

**Pull request merged**